File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        Integer var1 = x;
        String strX = var1.toString();

        int left = 0;
        int right = strX.length() -1;
        boolean isPalindrome = false;

        while (left <= right) {
            if (strX.charAt(left) != strX.charAt(right)) {
                return isPalindrome;
            }
            left++;
            right--;
        }

        isPalindrome = true;
        return isPalindrome;
    }
}

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        long long revNum = 0;
        long long original = x;
        int mod = 0;

        if (x < 0) return false;

        while(x != 0) {
            mod = x % 10;
            revNum = revNum * 10 + mod;
            x = x / 10;
        }

        if(revNum == original) return true;

        return false;
    }
};